docs/design/locale_fallback_and_negotiation.md

```diff
@@ -0,0 +1,427 @@
+
+Similar problems for all the languages used in many countries (date formats for “generic” Arabic, French, English, when there are tens of countries using these languages)
+
+I learned that the ICU matcher can return two results, one for the UI locale, one for i18n (formatting). ([`LocaleMatcher.Result.makeResolvedLocale`](https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/util/LocaleMatcher.Result.html#makeResolvedLocale--))
+
+That is, if you pass in:
+
+**Available UI:** [`es-419`, `es`]  
+**Desired:** [`es-CL`]
+
+You can get:
+
+**UI:** `es-419`  
+**i18n:** `es-CL`
+
+This is really nice, but still does not cover some valid situations well enough.
+
+For example Android would return `es-CL` in this case, even for UI. And the resource manager will do the fallback to `es-419` when loading the resources.
+
+The benefit: an application might have very-very few resources for `es-CL`. But they are still present.
+
+Some apps use this to keep things like country specific stuff (GPS coordinates, license agreement). But the bulk of the strings would be `es-419`.
+
+If you tell the resource manager to use `es-419` (the UI locale) then those country specific resources are not accessible.
+
+One option would be to pass every locale that has even one resource as “Available UI”:
+
+**Available UI:** [`es-419`, `es`, `es-CL`, `es-AR`, ...]
+**Desired:** [`es-CL`]
+
+Then the result would be:
+**UI:** `es-CL`
+**i18n:** `es-CL`
+
+And the resource manager will do the `es-CL` → `es-419` → `es` (vertical) fallback.
+
+But having a `LanguageMatcher` API that returns [`es-CL`, `es-419`, `es-AR`, `es`] can allow for a simpler resource manager implementation.
+The resource manager can use that API to implement that fallback on top of all the smartness that LanguageMatcher already has.
+
+**Conclusion**
+
+None of the two options is ideal, and I can’t think of a way to fix this with the public APIs provided by the current ICU LocaleMatcher.
+
+By providing a filtering method, similar to the current matching one, but returning a list of matches, then it would be possible to implement the more advanced behavior:
+
+* Take the comprehensive list of ICU4X supported locales  
+[ `af`, `af-NA`, `af-ZA`, …, **<span style="color:red">es, es-419, es-AR, …, es-CL, …, es-VE</span>**, …, **<span style="color:blue">ro, ro-MD, ro-RO</span>**, …, `zu`, `zu-ZA` ]
+* Filter it using the UI locale list (the current match, but returning a list)  
+[ `ar`, `es`, `es-419`, `en`, `en-GB`, `fr`, `zh` ]  
+⇒ [ `ar`, `ar-*`, `en`, `en-*`, `es`, `es-419`, `es-AR`, …, `es-CL`, …, `es-VE`, …, `fr`, `fr-*`, …, `zh`, `zh-Hans-*` ]  
+(the order of the result does not matter, but note how `zh-Hant` is not there)
+* Use this resulted (filtered) list as supported list, with the desired user list:  
+**supported:** [ `ar`, …, `es`, `es-419`, `es-AR`, …, `es-CL`, …, `es-VE`, …, `ro`, `ro-MD`, `ro-RO`, ... ]
+**desired:** [ `ro-RO`, `es-CL` ]  
```

Comment:
Put a newline before and after these 2 lines, to keep consistency with UI vs i18n and Available UI vs. Desired comparison excerpts in the rest of the document.

Reply:
Done